src/simplnx/Utilities/ThreadedAlgorithm.hpp

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include "simplnx/Filter/IFilter.hpp"
+#include "simplnx/simplnx_export.hpp"
+
+#include <chrono>
+#include <cstdint>
+
+namespace nx::core
+{
+
+class SIMPLNX_EXPORT ThreadedAlgorithm
+{
+
+public:
+  ThreadedAlgorithm(const IFilter::MessageHandler& mesgHandler, const std::atomic_bool& shouldCancel);
+  virtual ~ThreadedAlgorithm() noexcept;
+
+  ThreadedAlgorithm(const ThreadedAlgorithm&) = delete;
+  ThreadedAlgorithm(ThreadedAlgorithm&&) noexcept = delete;
+  ThreadedAlgorithm& operator=(const ThreadedAlgorithm&) = delete;
+  ThreadedAlgorithm& operator=(ThreadedAlgorithm&&) noexcept = delete;
+
+  const std::atomic_bool& getCancel() const;
+
+  void sendThreadSafeProgressMessage(size_t counter);
+
+  void sendThreadSafeInfoMessage(const std::string& message);
```

Comment:
```suggestion
 /**
   * @brief sets the message to be displayed
   * [!!Utilizes Mutex Locks!!] be wary calling this excessively in parallel as it can slow the program down
   * @param progressMessage display message
   */
  void sendThreadSafeInfoMessage(std::string&& message);
```
Documentation and swap to r-value, since the string should not be modified by calling function after it has been handed off. Move statements can be used to convert it if need be.